backbone.js

```diff
@@ -1487,11 +1483,12 @@
       this._wantsHashChange = this.options.hashChange !== false;
       this._hasHashChange   = 'onhashchange' in window;
       this._wantsPushState  = !!this.options.pushState;
-      this._hasPushState    = !!(this.options.pushState && this.history && this.history.pushState);
+      this._hasPushState    = !!(this.history && this.history.pushState);
+      this._usePushState    = this._wantsPushState && this._hasPushState;
```

Comment:
This is a gentle nudge towards https://github.com/jashkenas/backbone/pull/3394, so that the props actually mean what their name implies.